### ISSUE TITLE
Ktor plugin does not capture kotlinx rpc (and perhaps even websocket) traces.

### ISSUE THREAD

**Issue by jobinlawrance** [open-telemetry/opentelemetry-java-instrumentation#12939]
### Describe the bug

When adding opentelemetry plugin to ktor server, only the http request are captured in the traces. KotlinX RPC (krpc) is added to Ktor via a [plugin](https://ktor.io/docs/tutorial-first-steps-with-kotlin-rpc.html). 

Unlike gRPC, kotlinx RPC does not use Http 2 and instead relies on websockets as is evident from the logs and it's plugin implementation which also leads me to believe that websocket traces are missing as well.

```kotlin
// It's installed on the server like 

install(RPC)

// internal implementation

public val RPC: ApplicationPlugin<RPCConfigBuilder.Server> = createApplicationPlugin(
    name = "RPC",
    createConfiguration = { RPCConfigBuilder.Server() },
) {
    application.install(WebSockets)           // < -----------------------------  Websockets
    application.attributes.put(RPCServerPluginAttributesKey, pluginConfig)
}

// And on the client

installRPC {
            waitForServices = true
}

// implementation

public fun HttpClientConfig<*>.installRPC(
    configure: RPCConfigBuilder.Client.() -> Unit = {}
) {
    install(WebSockets)                      // < -----------------------------  Websockets
    install(RPC, configure)
}

```

### Steps to reproduce

In order to add OpenTelemetry plugin to Ktor server and client i've followed the offical [ktor examples](https://github.com/ktorio/ktor-samples/blob/main/opentelemetry/README.md) and implemented [here](https://github.com/jobinlawrance/zeroshare-multiplatform/commit/3a0e316ebe31c82af1a15c61086d2c69154f77e4)

### Expected behavior

The Ktor server has a bi-directional stream like 

```kotlin
routing {
        rpc("/stream") {
            rpcConfig {
                serialization {
                    json{
                        ignoreUnknownKeys = true
                    }
                }
            }
        
            registerService<DeviceStream> { ctx -> DeviceStreamImpl(ctx) }
        }
    }
```

and a http GET request like

```kotlin
routing {
        get("/") {
            call.respondText("Hello World!")
        }
    }
```

### Only the http requests are automatically captured on the server

<img width="1291" alt="image" src="https://github.com/user-attachments/assets/d05defcf-241d-4873-b664-ec8e33519a6c" />

<img width="1725" alt="image" src="https://github.com/user-attachments/assets/86429e6a-fb5c-4825-8696-7128541aa88a" />

### In client if the initial RPC call fails I do get the trace for it but otherwise there are no traces related to rpc & websockets

<img width="1711" alt="image" src="https://github.com/user-attachments/assets/d0c414f2-b2af-4d95-9706-ec5a8ce67767" />

### Javaagent or library instrumentation version

opentelemetry-ktor-3.0 v2.10.0-alpha

### Environment

**JDK**:
Temurin 21
**OS**:
MacOS Sequoia version 15.1

**Comment by laurit** [comment]
[We'd welcome a contribution for this](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/ISSUES.md#contribution-welcome).

**Comment by davidmrz** [comment]
Hi, I was exploring this repository and came across this issue that is open for contributions. I gave it a look, but wanted to understand deeper the issue reported.

I was able to reproduce it. No spans are registered for the `client` when calling via websockets (which RPC uses under the hood). However, the spans are registered for the server (not linked to the client). Would the expectation be that the trace includes both, client and server spans, or is it something else?

**Comment by laurit** [comment]
> Would the expectation be that the trace includes both, client and server spans, or is it something else?

yes, both client and server spans should be created and appear in the same trace.
